clients/cobol-lsp-vscode-extension/src/services/ProcessorGroupsLoader.ts

```diff
@@ -65,56 +64,65 @@ const ProcessorGroupModel = t.intersection([
 
 export type ProcessorGroup = t.TypeOf<typeof ProcessorGroupModel>;
 
-export const readProgramConfigFileContent = (): ProgramsConfig => {
+export async function readProgramConfigFileContent(
+  documentUri: Uri,
+): Promise<ProgramsConfig> {
   const EMPTY = { pgms: [] };
-  const ws = SettingsUtils.getWorkspaceFoldersPath(true);
-  if (ws.length < 1) {
-    return EMPTY;
-  }
-  const cfgPath = path.join(ws[0], PROCESSOR_GROUP_FOLDER);
-  const pgmCfgPath = path.join(cfgPath, PROCESSOR_GROUP_PGM);
-  if (!fs.existsSync(pgmCfgPath)) {
+
+  const ws = workspace.getWorkspaceFolder(documentUri);
+  if (ws === undefined) {
     return EMPTY;
   }
+  const cfgPath = Uri.joinPath(ws.uri, PROCESSOR_GROUP_FOLDER);
+  const pgmCfgPath = Uri.joinPath(cfgPath, PROCESSOR_GROUP_PGM);
```

Comment:
`joinPath` can actually join multiple segments... but I guess it will break all those mocks, right?